commands/package/list.js

```diff
@@ -0,0 +1,778 @@
+ * @returns {BlockFormatter} format version block
+ */
+function getVersionFormatter (options, paddings) {
+  return pkg => pkg.version.padEnd(paddings.get('padVersion'))
+}
+
+/**
+ * output a colored label in extended view
+ * @param {String} label the label
+ * @returns {String} colored terminal output
+ */
+function coloredLabel (label) {
+  return ct(label, 'FgGreen', 'Dim')
+}
+
+/**
+ * output authors with quantified label
+ * @param {ListOptions} options the options
+ * @returns {BlockFormatter} author formatter
+ */
+function getAuthorsFormatter (options) {
+  if (options.color) {
+    return pkg => coloredLabel(`Author${pkg.authors.length > 1 ? 's' : ''}: `) + pkg.authors.join(' ')
+  }
+  return pkg => `Author${pkg.authors.length > 1 ? 's' : ''}: ` + pkg.authors.join(' ')
+}
+
+/**
+ * shorten well-known processor URI
+ * @param {String} name processor URI
+ * @returns {String|"existdb"} maybe shortened name
+ */
+function processorName (name) {
+  return name === 'http://exist-db.org' ? 'existdb' : name
+}
+
+/**
+ * get processor block formatter
+ * @param {ListOptions} options the options
+ * @returns {BlockFormatter} the formatter
+ */
+function getProcessorFormatter (options) {
+  if (options.color) {
+    return function (pkg) {
+      if (!pkg.processor.uri) {
+        return coloredLabel('Processor: ') + 'any'
+      }
+      return coloredLabel('Processor: ') + processorName(pkg.processor.uri) + ' ' + ct(formatVersion(pkg.processor), 'FgYellow')
+    }
+  }
+  return function (pkg) {
+    if (!pkg.processor.name) {
+      return 'Processor: any'
+    }
+    return 'Processor: ' + processorName(pkg.processor.name) + ' ' + formatVersion(pkg.processor)
+  }
+}
+
+/**
+ * generic extended block formatter
+ * @param {ListOptions} options the options
+ * @param {String} label the label
+ * @param {String} prop the property
+ * @returns {BlockFormatter} the formatter
+ */
+function getLabelFormatter (options, label, prop) {
+  if (options.color) {
+    return pkg => (pkg[prop] ? coloredLabel(label + ': ') + pkg[prop] : null)
+  }
+  return pkg => (pkg[prop] ? label + ': ' + pkg[prop] : null)
+}
+
+/**
+ * helper function to render version ranges and templates
+ * @param {VersionedItem} versionedItem the versioned item
+ * @returns {String} the formatted version
+ */
+function formatVersion (versionedItem) {
+  if (versionedItem.exact.length) {
+    return versionedItem.exact.join(',')
+  }
+  if (versionedItem.template) {
+    const parts = versionedItem.template.split('.')
+    switch (parts.length) {
+      case 3: return versionedItem.template
+      case 2: return versionedItem.template + '.x'
+      default: return versionedItem.template + '.x.x'
+    }
+  }
+  if (versionedItem.min && versionedItem.max) {
+    return versionedItem.min + '-' + versionedItem.max
+  }
+  if (versionedItem.min) {
+    return '>=' + versionedItem.min
+  }
+  if (versionedItem.max) {
+    return '<=' + versionedItem.max
+  }
```

Comment:
Perhaps I'm reading this code wrong, but it appears that you're assuming that (a) only `@semver` can be an template and (b) `@semver-min` and `@semver-max` cannot be templates. But based on the EXPath Package spec's [section on templates](http://expath.org/spec/pkg#pkgdep), these attributes could all be templates. For a real life example of a `@semver-max` that could be a template, see https://github.com/eXist-db/public-repo/issues/72. Also, in this case, the rendering of `semver-max="5"` is `<6.0.0`, not `<=5.0.0`. See https://github.com/eXist-db/semver.xq/blob/master/src/main/xquery/semver.xqm#L208-L247.

Reply:
Very interesting find! Thanks for having a thorough look at it.

Reply:
I added unit tests for version checks and formatting to ensure it works and found additional problems.

Reply:
They are all fixed now with 5401b6c40634a90c5c7c0fef9cc0f6469fcb9e35

Reply:
`@semver-max="5"` translates to `<=5` now and the tests show that `5.99.999` satisfies.
The new function `formatVersionFill` will return `<=5.x.x` but this is not yet used anywhere. This could be made an option in `package list` in the future.

Reply:
While `<6.0.0` could be used and does transport the meaning, displaying this in the list could create confusion as the string "6" is not in the dependency declarations.

Reply:
@line-o  `@semver-max="5"` should translate to `<=6`, not `<=5`.

Reply:
(The one oddity that the EXPath Package spec authors may not have envisioned - since it predated SemVer 2.0 and maybe even 1.0? - is that `6.0.0-SNAPSHOT` satisfies `@semver-max=5`.)

Reply:
(EXPath Package "SemVer templates" are not particularly expressive. npm's SemVer ranges are much more. But they're also [not in the SemVer spec](https://github.com/semver/semver/issues/205). It would be nice to have a spec to reference when considering a successor to EXPath Package SemVer templates for an EXPath Package 2.0... In the absence of this, the EXPath Package spec and SemVer 2.0 is all we've really got.)

Reply:
Whereas `6.0.0-SNAPSHOT` does __not__  satisfy `<=5`. The intended outcome. @joewiz I would really like to stick to npm SemVer ranges in the output as this is - in my mind - the best we have.

Reply:
Can you point me to the resource in either EXPath Package 2.0 or SemVer 2.0 from where I can see that  `semver-max="5"` translates to `<6`?
If this is implemented in that way in the public repo, I would consider it an issue.

Reply:
> (The one oddity that the EXPath Package spec authors may not have envisioned - since it predated SemVer 2.0 and maybe even 1.0? - is that 6.0.0-SNAPSHOT satisfies @semver-max=5.)

@joewiz Can you please explain how you got to this conclusion?

Reply:
The expath spec doesn’t mention pre-release labels. So my reading is that a template `5` means anything with a 5 major version. If we want to include `5.0.0-Snapshot` or `6.0.0-Snapshot` in that is up to us. In light of the communities development practices I m against including the latter. As per SemVer definition, it includes breaking aka incompatible changes. I m not happy with including the former but I can live with it. 

TLDR; +1 for `<=`